hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/web/resources/NamenodeWebHdfsMethods.java

```diff
@@ -288,18 +292,18 @@ static DatanodeInfo chooseDatanode(final NameNode namenode,
       throw new IOException("Namesystem has not been initialized yet.");
     }
     final BlockManager bm = fsn.getBlockManager();
-    
-    HashSet<Node> excludes = new HashSet<Node>();
+
+    Set<Node> excludes = new HashSet<>();
     if (excludeDatanodes != null) {
       for (String host : StringUtils
           .getTrimmedStringCollection(excludeDatanodes)) {
-        int idx = host.indexOf(":");
+        int idx = host.indexOf(':');
         Node excludeNode = null;
-        if (idx != -1) {
```

Comment:
This makes the code easier to read by just switching the `!=` with the `==` condition. No other code changes